### PR TITLE
Include ecog and seeg

### PR DESCRIPTION
**Pull request by chapochn**
the function get_rejection_threshold now also works when the data is ecog or seeg

### CODE REVIEW

**Comment by chapochn** [comment]
I don't really understand the error

**Comment by chapochn** [comment]
related to: https://github.com/autoreject/autoreject/issues/276

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -258,6 +259,10 @@ def get_rejection_threshold(epochs, decim=1, random_state=None,
             picks = pick_types(epochs.info, meg='grad', eeg=False)
         elif ch_type in ['hbo', 'hbr']:
             picks = pick_types(epochs.info, meg=False, fnirs=ch_type)
+        elif ch_type == 'ecog':
+            picks = pick_types(epochs.info, ecog=True)
+        elif ch_type == 'seeg':
+            picks = pick_types(epochs.info, seeg=True)
```
Comment (jasmainak) [inline autoreject/autoreject.py:265]: is there a tiny test you could add?

**Comment by sappelhoff** [comment]
No idea ... I was hoping this was a temporary issue. But it seems to persist. Somebody will have to dig into this :-)

**Comment by hoechenberger** [comment]
Seems like yet another issue with OpenNeuro. We'll have to
file a bug report...

**Comment by sappelhoff** [comment]
@chapochn would you mind filing a bug report about this on https://github.com/OpenNeuroOrg/openneuro/ ?

**Comment by chapochn** [comment]
@sappelhoff I have no experience with OpenNeuro.
This is the error from circleci:
```

Sphinx-Gallery successfully executed 6 out of 7 files subselected by:

    gallery_conf["filename_pattern"] = '/plot'
    gallery_conf["ignore_pattern"]   = '__init__\\.py'

after excluding 0 files that had previously been run (based on MD5).


Traceback (most recent call last):
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app.build(args.force_all, filenames)
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/sphinx/application.py", line 331, in build
    self.events.emit('build-finished', None)
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/sphinx/events.py", line 94, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/sphinx_gallery/gen_gallery.py", line 732, in summarize_failing_examples
    raise ExtensionError(fail_message)
sphinx.errors.ExtensionError: Here is a summary of the problems encountered when running the examples

Unexpected failing examples:
/home/circleci/project/examples/plot_visualize_bad_epochs.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/circleci/project/examples/plot_visualize_bad_epochs.py", line 33, in <module>
    openneuro.download(dataset=dataset, target_dir=target_dir,
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/openneuro/download.py", line 684, in download
    asyncio.run(_download_files(**kwargs))
  File "/opt/circleci/.pyenv/versions/3.9.7/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/circleci/.pyenv/versions/3.9.7/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/openneuro/download.py", line 468, in _download_files
    await asyncio.gather(*download_tasks)
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/openneuro/download.py", line 330, in _download_file
    raise RuntimeError(
RuntimeError: Error 404 when trying to download /home/circleci/project/autoreject/../examples/ds000117/dataset_description.json from https://s3.amazonaws.com/openneuro.org/ds000117/dataset_description.json?versionId=vbCEkbZtCL26q._PCPmrCJK_KngbQ_dT


-------------------------------------------------------------------------------

Extension error:
Here is a summary of the problems encountered when running the examples

Unexpected failing examples:
/home/circleci/project/examples/plot_visualize_bad_epochs.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/circleci/project/examples/plot_visualize_bad_epochs.py", line 33, in <module>
    openneuro.download(dataset=dataset, target_dir=target_dir,
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/openneuro/download.py", line 684, in download
    asyncio.run(_download_files(**kwargs))
  File "/opt/circleci/.pyenv/versions/3.9.7/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/circleci/.pyenv/versions/3.9.7/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/openneuro/download.py", line 468, in _download_files
    await asyncio.gather(*download_tasks)
  File "/home/circleci/autoreject_env/lib/python3.9/site-packages/openneuro/download.py", line 330, in _download_file
    raise RuntimeError(
RuntimeError: Error 404 when trying to download /home/circleci/project/autoreject/../examples/ds000117/dataset_description.json from https://s3.amazonaws.com/openneuro.org/ds000117/dataset_description.json?versionId=vbCEkbZtCL26q._PCPmrCJK_KngbQ_dT


-------------------------------------------------------------------------------
make: *** [Makefile:20: html] Error 2

Exited with code exit status 2

CircleCI received exit code 2

```

What is the command that is causing the error?

File: autoreject/autoreject.py (hunk 1 above)
Comment (chapochn) [reply autoreject/autoreject.py:265]: let me look into that

**Comment by hoechenberger** [comment]
I've field a bug report at https://github.com/OpenNeuroOrg/openneuro/issues/2635

**Comment by sappelhoff** [comment]
and apparently already solved -- I'll restart CI

**Comment by chapochn** [comment]
I added 2 tests, but I can see that some checks failed. I'm not sure how to see the error, do I need to be able to access Azure? I don't seem to have access. Please let me know.

**Comment by sappelhoff** [comment]
yeah, I need to sign in with my microsoft account -- and I don't have access to the logs: 
image

File: autoreject/tests/test_autoreject.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 from mne.datasets import sample
 from mne import io
 from mne.utils import _TempDir
+from mne_bids import BIDSPath, read_raw_bids
```
Comment (jasmainak) [inline autoreject/tests/test_autoreject.py:21]: this is causing the CI to fail.

**Comment by jasmainak** [comment]
Can you just change the channel type of a regular EEG channel so as to not add a dependency on `mne_bids` for the test? There is nothing specific about sEEG in your test at the moment.

File: autoreject/tests/test_autoreject.py (hunk 2 above)
Comment (chapochn) [reply autoreject/tests/test_autoreject.py]: thanks, I adjusted

File: autoreject/tests/test_autoreject.py [hunk 3]
```diff
@@ -364,3 +364,60 @@ def test_fnirs():
...
+    print(reject)
+    epochs.drop_bad(reject=reject)
+    n2 = len(epochs)
+    assert "ecog" in reject.keys()
+    assert reject["ecog"] > 0.0
+    assert reject["ecog"] < 0.01
+    assert n2 < n1
+
+
+def test_seeg():
```
Comment (jasmainak) [inline autoreject/tests/test_autoreject.py:399]: can you avoid the code duplication and combine the two tests?

File: autoreject/tests/test_autoreject.py [hunk 4]
```diff
@@ -364,3 +364,60 @@ def test_fnirs():
     assert reject["hbo"] < 0.001  # This is a very high value as sanity check
     assert reject["hbr"] < 0.001
     assert reject["hbr"] > 0.0
+
+
+def test_ecog():
```
Comment (jasmainak) [inline autoreject/tests/test_autoreject.py:369]: we follow PEP 257. See how docstrings should be formatted: https://peps.python.org/pep-0257/

File: autoreject/tests/test_autoreject.py (hunk 3 above)
Comment (chapochn) [reply autoreject/tests/test_autoreject.py:399]: ok done

File: autoreject/tests/test_autoreject.py (hunk 4 above)
Comment (chapochn) [reply autoreject/tests/test_autoreject.py:369]: not sure to which place you're referring to, but I changed something, see in the next commit

Comment (jasmainak) [reply autoreject/tests/test_autoreject.py:369]: see: "Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description."

Comment (chapochn) [reply autoreject/tests/test_autoreject.py:369]: ok thanks, pushed a new version

File: autoreject/tests/test_autoreject.py [hunk 5]
```diff
@@ -364,3 +364,29 @@ def test_fnirs():
...
+
+        # make events
+        events = np.arange(1315, 1365, 2, dtype=int) * 1000  # in samples
+        events = np.array([[e, 0, 0] for e in events])
+        epochs = mne.Epochs(raw, events, event_id=0,
+                            tmin=-1, tmax=1, baseline=None, verbose=True)
+        epochs.load_data()
+        n1 = len(epochs)
+        reject = get_rejection_threshold(epochs)
+        print(reject)
```
Comment (jasmainak) [inline autoreject/tests/test_autoreject.py:386]:
```suggestion
```

no print statements in tests ...

**Comment by jasmainak** [comment]
Can you update `whats_new.rst` so we can give you credit in the next release?

File: autoreject/tests/test_autoreject.py (hunk 5 above)
Comment (chapochn) [reply autoreject/tests/test_autoreject.py:386]: ok, I copied that from a test that was already there

Comment (chapochn) [reply autoreject/tests/test_autoreject.py]: pushed

**Comment by chapochn** [comment]
ok, updated changelog, thanks for suggesting!

**Comment by jasmainak** [comment]
looks like you need to rebase with respect to the latest master branch:

```sh
$ git fetch upstream master:master
$ git rebase master
$ git push -f origin include-ecog
```

**Comment by chapochn** [comment]
Should be ok now

**Comment by jasmainak** [comment]
image

I still can't merge ...

**Comment by jasmainak** [comment]
I went ahead and squashed your commits. Thanks @chapochn ! 🥳

**Comment by chapochn** [comment]
Great, thank you!

**Pull request merged**